pkg/controller/mizar/grpc-adaptor.go

```diff
@@ -0,0 +1,272 @@
+/*
+Copyright 2020 Authors of Arktos.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mizar
+
+import (
+	"context"
+	fmt "fmt"
+	"time"
+
+	"google.golang.org/grpc"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	port = "50052"
+)
+
+// GrpcCreateService is to invoking grpc func of CreateService
```

Comment:
Guess we can have simpler code by abstrction using func as a param. For example, for those Service related functions:

func GrpcServiceOperation(grpcHost string, service *v1.Service, operation func(context.Context, *BuiltinsServiceMessage) (*ReturnCode, erroRr) ) *ReturnCode {
	client, ctx, conn, 
        cancel, err := getGrpcClient(grpcHost)
	if err != nil {
		return getReturnCodeFromError(&err)
	}
	defer conn.Close()
	defer cancel()
	returnCode, err := operation(ctx, ConvertToServiceContract(service))
	if err != nil {
		return getReturnCodeFromError(&err)
	}
	return returnCode
}

func GrpcCreateService(grpcHost string, service *v1.Service) *ReturnCode {
       return GrpcServiceOperation(grpcHost, service, client.CreateService)
}

...

Reply:
Actually, more abstraction can be done over the different types of resources. Yet it involves more type conversion and check, I had a brief look and it seems to me that the signature of those conversion funcs need to be updated to handler the possible conversion errors.

So I just stop here, you can play with it more to have even simpler code.